### PR TITLE
Make casper replay on top of the least effort block

### PR DESCRIPTION
**Pull request by itegulov**
## Overview
<sup>_What this PR does, and why it's needed_</sup>

### JIRA ticket:
<sup>_Create it if there isn't one already._</sup>

### Notes
<sup>_Optional. Add any notes on caveats, approaches you tried that didn't work, or anything else._</sup>

### Please make sure that this PR:
- [x] is at most 200 lines of code (excluding tests),
- [x] meets [RChain development coding standards](https://rchain.atlassian.net/wiki/spaces/DOC/pages/28082177/Coding+Standards),
- [x] includes tests for all added features,
- [x] has a reviewer assigned,
- [x] has [all commits signed](https://rchain.atlassian.net/wiki/spaces/DOC/pages/498630673/How+to+sign+commits+to+rchain+rchain).

### [Bors](https://bors.tech/) cheat-sheet:

- `bors r+` runs integration tests and merges the PR (if it's approved),
- `bors try` runs integration tests for the PR,
- `bors delegate+` enables non-maintainer PR authors to run the above.

### CODE REVIEW

**Comment by 9rb** [comment]
bors try

**Comment by 9rb** [comment]
bors try

**Review by 9rb — APPROVED** [review]

File: casper/src/main/scala/coop/rchain/casper/util/rholang/InterpreterUtil.scala [hunk 1]
```diff
@@ -208,8 +208,17 @@ object InterpreterUtil {
       case Seq((_, parentStateHash)) =>
         parentStateHash.pure
 
-      case (_, initStateHash) +: _ =>
-        replayIntoMergeBlock(parents, dag, runtimeManager, initStateHash)
```
Comment (tgrospic) [inline casper/src/main/scala/coop/rchain/casper/util/rholang/InterpreterUtil.scala:212]: Is this function (replayIntoMergeBlock) used anywhere else?

**Review by tgrospic — APPROVED** [review]
Please create a ticket for tests to cover this new functionality.

File: casper/src/main/scala/coop/rchain/casper/util/rholang/InterpreterUtil.scala (hunk 1 above)
Comment (nzpr) [reply casper/src/main/scala/coop/rchain/casper/util/rholang/InterpreterUtil.scala]: https://rchain.atlassian.net/browse/RCHAIN-4016

Comment (nzpr) [reply casper/src/main/scala/coop/rchain/casper/util/rholang/InterpreterUtil.scala:212]: @tgrospic no, this is the only place where it is called.
Ticket created https://rchain.atlassian.net/browse/RCHAIN-4016.

**Pull request merged**